src/hyperkit.c

```diff
@@ -745,51 +745,44 @@ firmware_parse(const char *opt) {
 		goto fail;
 	}
 
-		if ((cp = strchr(fw, ',')) != NULL) {
-			*cp = '\0';
-			opt1 = cp + 1;
-		} else {
-			goto fail;
-		}
-
-		if ((cp = strchr(opt1, ',')) != NULL) {
-			*cp = '\0';
-			opt2 = cp + 1;
-		} else {
-			goto fail;
-		}
+// Gets next comma-separated option and stores it in next.
+// If the length of the option is 0 it is replaced with NULL.
+#define NEXTARG(cur, next, scratch) do {			\
+	if (cur && (scratch = strchr(cur, ',')) != NULL) {	\
+		*scratch = '\0';				\
+		next = scratch + 1;				\
+		next = next && strlen(next) ? next : NULL;	\
```

Comment:
`next` cannot be `NULL` here (exception some sort of wraparound in `scratch + 1`.

Also wont setting `next = NULL` here cause the next invocation of `NEXTARG` to fail the initial condition and therefore break the `foo,,bar` case because `opt2` ends up NULL and hence `NEXTARGS(opt2, opt3, cp)` fails the `if (cur`.